sootup.core/src/main/java/sootup/core/validation/InvokeArgumentValidator.java

```diff
@@ -22,31 +22,72 @@
  * #L%
  */
 
-import java.util.List;
+import sootup.core.IdentifierFactory;
+import sootup.core.jimple.basic.Immediate;
+import sootup.core.jimple.common.expr.AbstractInvokeExpr;
+import sootup.core.jimple.common.stmt.Stmt;
 import sootup.core.model.Body;
+import sootup.core.signatures.MethodSignature;
+import sootup.core.typehierarchy.TypeHierarchy;
+import sootup.core.types.ClassType;
+import sootup.core.types.PrimitiveType;
+import sootup.core.types.Type;
 import sootup.core.views.View;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
 /**
  * A basic validator that checks whether the length of the invoke statement's argument list matches
  * the length of the target methods's parameter type list.
  *
  * @author Steven Arzt
  */
 public class InvokeArgumentValidator implements BodyValidator {
+    @Override
+    public List<ValidationException> validate(Body body, View view) {
+        List<ValidationException> validationException = new ArrayList<>();
+
+        for (Stmt stmt : body.getStmts()) {
+            if (stmt.containsInvokeExpr()) {
+                AbstractInvokeExpr invExpr =
+                        stmt.getInvokeExpr();
+                MethodSignature callee = invExpr.getMethodSignature();
+                List<Immediate> args = invExpr.getArgs();
+                List<Type> parameterTypes = callee.getParameterTypes();
+                if (invExpr.getArgCount() != parameterTypes.size()) {
+                    validationException.add(new ValidationException(stmt, "Invalid number of arguments"));
+                } else {
+                    // check argument type
+                    TypeHierarchy typeHierarchy = view.getTypeHierarchy();
+                    IdentifierFactory identifierFactory = view.getIdentifierFactory();
+                    Iterator<Immediate> iterArgs = args.iterator();
+                    Iterator<Type> iterParameters = parameterTypes.iterator();
+                    while (iterArgs.hasNext() && iterParameters.hasNext()) {
+                        ClassType argType = getClassType(identifierFactory, iterArgs.next().getType());
+                        ClassType paramType = getClassType(identifierFactory, iterParameters.next());
+                        if (argType != paramType && (!typeHierarchy.contains(paramType)
```

Comment:
I am not sure the Typehierarchy can handle primitive type relations / BoxedType compatibility - maybe we have to do that seperately via [type promotion](https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html)

Reply:
But this does not handle cases like: `int` is used as an argument of a `double` parameter (while double -> int is not allowed implicitly) - we should not fail in these cases